02_selection_sort/java/src/SelectionSort2.java

```diff
@@ -3,45 +3,23 @@
 public class SelectionSort2 {
 
     // this version uses raw arrays instead of ArrayList
-    private static int[] selectionSort(int[] arr) {
-        int[] newArr = new int[arr.length];
-
-        for (int i = 0; i < newArr.length; i++) {
-            int smallestIndex = findSmallest(arr);
-            newArr[i] = arr[smallestIndex];
-
-            arr = getNewArrWithoutSmallest(arr, smallestIndex);
-        }
-
-        return newArr;
-    }
-
-    private static int[] getNewArrWithoutSmallest(int[] arr, int smallestIndex) {
-        int[] newArrWithoutSmallest = new int[arr.length - 1];
-        for (int i = 0; i < arr.length; i++) {
-            if (i < smallestIndex) {
-                newArrWithoutSmallest[i] = arr[i];
-            } else if (i > smallestIndex) {
-                newArrWithoutSmallest[i - 1] = arr[i];
-            }
-        }
-        return newArrWithoutSmallest;
-    }
-
-    private static int findSmallest(int[] arr) {
-        int smallest = arr[0];
-        int smallestIndex = 0;
-        for (int i = 0; i < arr.length; i++) {
-            if (arr[i] < smallest) {
-                smallest = arr[i];
-                smallestIndex = i;
+    public static void selectionSort(int[] target) {
+        for (int i = 0; i < target.length - 1; i++) {
+            int left = target[i];
+            for (int j = i + 1; j < target.length; j++) {
+                int rigth = target[j];
```

Comment:
please change `rigth` to `right` and I'll merge

Reply:
+

Reply:
Done